feat: use tabular numbers in frontend

FlorianLudwig opened the pull request:
## Proposed change

This makes all digits monospaced, improving readability of date columns.

[Screencast from 2023-12-28 16-39-48.webm](https://github.com/paperless-ngx/paperless-ngx/assets/432263/ab7e8030-b360-4dff-82fb-07c302909e08)


## Type of change

<!--
What type of change does your PR introduce to Paperless-ngx?
NOTE: Please check only one box!
-->

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] (Small) New feature: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

<!--
NOTE: PRs that do not address the following will not be merged, please do not skip any relevant items.
-->

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

FlorianLudwig committed:
feat: use tabular numbers in frontend

This makes all digits monospaced, improving readability of date columns.

shamoon commented:
Thanks

Not to quibble here but it does seem kinda subjective. Since we're not using numbers in a way where they are getting added (like a list of monetary sums) this doesnt seem perfectly applicable, so it's kind of just a typographic opinion. And typographically, I dont prefer this overall. I do think it's kind of aesthetically pleasing to see things line up, but in many instances it just looks like someone set the kerning wrong.

Again, its pretty subjective, so you can open a FR and if theres a lot of interest in this we can readdress.

<img width="1840" alt="Screenshot 2023-12-28 at 8 04 46 AM" src="https://github.com/paperless-ngx/paperless-ngx/assets/4887959/efb67c56-0d68-42c9-bd01-d772c72bf7d9">
<img width="1840" alt="Screenshot 2023-12-28 at 8 04 58 AM" src="https://github.com/paperless-ngx/paperless-ngx/assets/4887959/72162329-0ff3-452f-87e6-3d4b7b566350">

Ps. I think its also exaggerated by the font youre using, is the a default on some system / browser? Which one?

FlorianLudwig commented:
@shamoon thank you for the quick and detailed response!

I did not notice that the default font is used and it looks so different between systems. So I changed my default font :sweat_smile:  Thanks for the hint.

I do think it would still be nice to set the date columns to tabular digits but that is not as quick for me to implement. So I added it globally via tampermonkey for me :)
